### ISSUE TITLE
Fails on ubuntu 16.04.1

### ISSUE THREAD

**Issue by JeffAbrahamson** [saz/puppet-locales#29]
Asking puppet apply to do this

```
class english {

  class { 'locales':
default_locale  => 'en_GB.UTF-8',
locales         => ['en_GB.UTF-8 UTF-8', 'en_US.UTF-8 UTF-8'],
  }
}
```

leads to death thus:

```
Error: Could not set 'file' on ensure: No such file or directory @ dir_s_rmdir - /var/lib/locales/supported.d/local20161004-32612-1qyuldy.lock at 191:/etc/puppet/modules/locales/manifests/init.pp
Error: Could not set 'file' on ensure: No such file or directory @ dir_s_rmdir - /var/lib/locales/supported.d/local20161004-32612-1qyuldy.lock at 191:/etc/puppet/modules/locales/manifests/init.pp
Wrapped exception:
No such file or directory @ dir_s_rmdir - /var/lib/locales/supported.d/local20161004-32612-1qyuldy.lock
Error: /Stage[main]/Locales/File[/var/lib/locales/supported.d/local]/ensure: change from absent to file failed: Could not set 'file' on ensure: No such file or directory @ dir_s_rmdir - /var/lib/locales/supported.d/local20161004-32612-1qyuldy.lock at 191:/etc/puppet/modules/locales/manifests/init.pp
Notice: /Stage[main]/Locales/Exec[locale-gen]: Dependency File[/var/lib/locales/supported.d/local] has failures: true
Warning: /Stage[main]/Locales/Exec[locale-gen]: Skipping because of failed dependencies
```

The missing file `/var/lib/locales/supported.d` is probably a red herring, since it appears to be create on successful locale generation (cf. [here](https://help.ubuntu.com/community/Locale)).

The bug appears to exist both in the version pulled from [puppet forge](https://forge.puppet.com/saz/locales) and that from github.

I'm using puppet 3.8.5.

**Comment by JeffAbrahamson** [comment]
A bi t more info:

For whatever reason, the directory `var/lib/locales` is not created on 16.04.  Yet the package works on a fresh 15.10 install, where that file exists but is not installed by dpkg.

    vagrant@seine:/vagrant$ dpkg -S /var/lib/locales
    dpkg-query: no path found matching pattern /var/lib/locales
    vagrant@seine:/vagrant$

What is failing is this snippet:

    file { $config_file:
      ensure  => $config_ensure,
      owner   => 'root',
      group   => 'root',
      mode    => '0644',
      require => Package[$package],
      notify  => Exec['locale-gen'],
    }

because `$config_file` is `/var/lib/locales/supported.d/local`.

It looks like `/var/lib/locales/supported.d/local` is no longer created [cf. [issue](https://github.com/ronnix/fabtools/issues/325)].  It looks like a fix is to check `$::operatingsystemrelease`.  I'll prepare a patch.

**Comment by JeffAbrahamson** [comment]
Pull request [here](https://github.com/saz/puppet-locales/pull/30).

I've tested simply by running `sudo puppet apply  --modulepath=/etc/puppet/modules --execute "class { 'locales': default_locale  => 'en_GB.UTF-8', locales         => ['en_GB.UTF-8 UTF-8', 'en_US.UTF-8 UTF-8'], }"` on an ubuntu 15.10 vagrant box and on an ubuntu 16.04 vagrant box.  I see that there's a rudimentary test that doesn't do more than that, though maybe it's invokable more easily.

Feedback welcome, of course.

**Comment by HielkeJ** [comment]
The pull request works for me. Is it possible to create a new version and update it on the forge?

**Comment by zdavis** [comment]
Pull request works for us as well. Would be great to see this released!

**Comment by saz** [comment]
This should be fixed with the newest release v2.3.0
